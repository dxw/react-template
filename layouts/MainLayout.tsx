import Head from "next/head";
import React, { ReactNode } from "react";

type LayoutProps = {
  children: ReactNode;
};

const MainLayout = ({ children }: LayoutProps): JSX.Element => (
  <>
    <Head>
      <title>
        {/* TODO: Replace this with the base title for the service. */}
        React template
      </title>
    </Head>

    {children}

    {/* TODO: Replace the contents of the `style` tag with any global styles, or remove it if it's not applicable. */}
    <style jsx global>
      {`
        body {
          font-family: "Lucida Sans", "Lucida Sans Regular", "Lucida Grande",
            "Lucida Sans Unicode", Geneva, Verdana, sans-serif;
        }
      `}
    </style>
  </>
);

export default MainLayout;
