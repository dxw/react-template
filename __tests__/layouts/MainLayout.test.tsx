/* eslint-env jest */
import React from "react";
import { create } from "react-test-renderer";

import MainLayout from "../../layouts/MainLayout";

it("renders correctly", () => {
  const component = create(
    <MainLayout>
      <p>Test content</p>
    </MainLayout>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
