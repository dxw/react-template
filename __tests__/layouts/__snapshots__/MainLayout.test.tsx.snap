// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <p>
    Test content
  </p>,
  <style
    global={true}
    jsx={true}
  >
    
        body {
          font-family: "Lucida Sans", "Lucida Sans Regular", "Lucida Grande",
            "Lucida Sans Unicode", Geneva, Verdana, sans-serif;
        }
      
  </style>,
]
`;
